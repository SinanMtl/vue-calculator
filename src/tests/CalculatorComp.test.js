import { describe, expect, it } from 'vitest';
import { mount, flushPromises } from '@vue/test-utils';
import Calculator from '../components/CalculatorComp.vue';

describe('Calculator component', () => {
  const wrapper = mount(Calculator);

  it('should be mounted', () => {
    expect(wrapper.html()).toMatchSnapshot();
  })

  it('should be clicked any button', async () => {
    const btn = wrapper.find('[data-test="5"]');
    const screen = wrapper.find('[data-test="screen"]');

    await btn.trigger('click');
    await flushPromises();

    expect(screen.text()).toBe('5');
  });

  it('add 3', async () => {
    const btnAdd = wrapper.find('[data-test="+"]');
    const btn3 = wrapper.find('[data-test="3"]');
    const btnFinish = wrapper.find('[data-test="="]');
    const history = wrapper.find('[data-test="history"]');
    const screen = wrapper.find('[data-test="screen"]');

    await btnAdd.trigger('click');
    await btn3.trigger('click');
    await btnFinish.trigger('click');
    await flushPromises();

    expect(history.text()).toBe('5 + 3');
    expect(screen.text()).toBe('8');
  });
});
