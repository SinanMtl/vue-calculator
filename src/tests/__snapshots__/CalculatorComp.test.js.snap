// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Calculator component > should be mounted 1`] = `
"<div class=\\"bg-black px-[18px] py-[22px] pt-[90px]\\">
  <div data-test=\\"history\\" class=\\"text-right font-sans font-light text-[37px] text-gray-600\\"></div>
  <div data-test=\\"screen\\" class=\\"mb-7 font-worksans font-light text-[62px] leading-none text-right\\"></div>
  <div class=\\"grid gap-2.5 grid-cols-4\\"><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\" data-test=\\"C\\">C</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\" data-test=\\"S\\"><img src=\\"/sign.svg\\" alt=\\"Sign\\"></button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\" data-test=\\"%\\">%</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center bg-orange-500 hover:bg-orange-800 pb-[7px]\\" data-test=\\"/\\">÷</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\" data-test=\\"7\\">7</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\" data-test=\\"8\\">8</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\" data-test=\\"9\\">9</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center bg-orange-500 hover:bg-orange-800 pb-[7px]\\" data-test=\\"*\\">×</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\" data-test=\\"4\\">4</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\" data-test=\\"5\\">5</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\" data-test=\\"6\\">6</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center bg-orange-500 hover:bg-orange-800 pb-[7px]\\" data-test=\\"-\\">-</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\" data-test=\\"1\\">1</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\" data-test=\\"2\\">2</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\" data-test=\\"3\\">3</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center bg-orange-500 hover:bg-orange-800 pb-[7px]\\" data-test=\\"+\\">+</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center text-[40px] indent-[-7px]\\" data-test=\\"D\\">⌫</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\" data-test=\\"0\\">0</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\" data-test=\\".\\">,</button><button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center bg-orange-500 hover:bg-orange-800 pb-[7px]\\" data-test=\\"=\\">=</button></div>
</div>"
`;
